### DISCUSSION TITLE
Getting maximum number of clients error when using Drawable Text

### DISCUSSION THREAD

**Discussion by JJunHan** [ImageMagick/ImageMagick#5598] [Help]
Hi, its my first time using imagemagick and i am attempting to draw on multiple images in a for-loop and display it using SXIV viewer.

However, i am getting this error: 
"Maximum number of clients reachedMagick: unable to open X server '(null)': Resource temporarily unavailable @ error/xwindow.c/XrenderImage/9059.N6Magick9ErrorDrawE"

This error occurs after multiple successful drawings, and is temporarily solved by restarting the program. It does not occur if i totally remove the drawing of text onto the image. (Drawing rectangles doesnt cause this problem, only drawingtext : using DrawableText)

```
pid = fork();
if (pid == 0) {
execl("/usr/bin/sxiv", "sxiv", "sample", (char *)0);
exit(1);
}

while(1){
Blob blob( &imgData[0], sizeof(uint8)*imgData.size() );
Image image;
image.size("640x480");
image.magick("JPEG");
image.read(blob);
std::vector<Magick::Drawable> textlist;
std::vector<Magick::Drawable> drawList;
// Add some drawing options to drawing list
textlist.push_back(DrawableFont("-misc-fixed-medium-r-normal--20-200-75-75-c-100-iso8859-1"));
textlist.push_back(DrawableStrokeColor("yellow")); // Outline color
textlist.push_back(DrawableFillColor("yellow")); // Fill color
drawList.push_back(DrawableStrokeColor("red")); // Outline color
drawList.push_back(DrawableStrokeWidth(5)); // Stroke width
drawList.push_back(DrawableFillColor("none")); // Fill color
for(int j=0;j<count*6;j=j+6){
			string score_str = "Score: "+ to_string(bboxdata[4+j]) + " Class: " + class_name;
			textlist.push_back(DrawableText(bboxdata[0+j], bboxdata[1+j], score_str));
			drawList.push_back(DrawableRectangle(bboxdata[0+j],bboxdata[1+j],bboxdata[0+j]+bboxdata[2+j],bboxdata[1+j]+bboxdata[3+j]));
}
image.draw(drawList);
image.draw(textlist);
Image temp_image(image); 
temp_image.write("sample");
}
```

The code snippet above will eventually lead to that error mentioned. However, removing the textlist stops the error. Is there anyway to fix this? Thank you so much. Any advice would be greatly appreciated!

**Comment by snibgo** [comment]
I can't see anything in your code that tries to open an X server. Nor is there anything that declares "image" to which ".draw" could apply.

I suggest you show a complete but minimal program that has the error.

However, I don't use X windows, so I am unlikely to be able to help.

**Reply by JJunHan** [reply]
Hi, i have updated my code snippet. I forked a sxiv viewer to read the image that was drawn on and saved.

I am puzzled too because i dont see dont what causes that error but im sure it has something to do with DrawableText.
